t/40-script_openqa-clone-custom-git-refspec.t

```diff
@@ -46,21 +45,33 @@ test_once '--invalid-arg', qr/Usage:/, 'invalid args also yield help', 0, 'help
 my $args = 'https://github.com/user/repo/pull/9128 https://openqa.opensuse.org/tests/1234';
 isnt run_once($args), 0, 'without network we fail (without error)';
 # mock any external access with all arguments
+
+my $branch = "my_branch";
 $ENV{curl_github} = qq{echo -e '{"head": {"label": "user:my_branch"}, "body": "Lorem ipsum"}'; true};
 $ENV{curl_openqa}
   = qq{echo -e '{"TEST": "my_test", "CASEDIR": "/my/case/dir", "PRODUCTDIR": "/my/case/dir/product", "NEEDLES_DIR": "/my/case/dir/product/needles"}'; true};
 my $clone_job = 'openqa-clone-job --skip-chained-deps --within-instance https://openqa.opensuse.org ';
 my $dirs
-  = 'CASEDIR=https://github.com/user/repo.git#my_branch PRODUCTDIR=repo/product NEEDLES_DIR=/my/case/dir/product/needles';
-my $expected    = $clone_job . '1234 _GROUP=0 TEST=my_test@user/repo#my_branch BUILD=user/repo#9128 ' . $dirs;
+  = "CASEDIR=https://github.com/user/repo.git#$branch PRODUCTDIR=repo/product NEEDLES_DIR=/my/case/dir/product/needles";
+my $expected    = $clone_job . '1234 _GROUP=0 TEST=my_test@user/repo#' . $branch . ' BUILD=user/repo#9128 ' . $dirs;
 my $expected_re = qr/${expected}/;
 test_once $args, $expected_re, 'clone-job command line is correct';
 test_once "-v $args",    qr/\+ local dry_run/, 'clone-job with -v prints commands';
 test_once "-n -v $args", qr/\+ local dry_run/, 'clone-job with -n -v prints commands';
-my $args_branch = 'https://github.com/user/repo/tree/my_branch https://openqa.opensuse.org/tests/1234 FOO=bar';
+my $args_branch = "https://github.com/user/repo/tree/$branch https://openqa.opensuse.org/tests/1234 FOO=bar";
+
 my $expected_branch_re
-  = qr{${clone_job}1234 _GROUP=0 TEST=my_test\@user/repo#my_branch BUILD=user/repo#my_branch ${dirs} FOO=bar};
+  = qr{${clone_job}1234 _GROUP=0 TEST=my_test\@user/repo#$branch BUILD=user/repo#$branch ${dirs} FOO=bar};
 test_once $args_branch, $expected_branch_re, 'alternative mode with branch reference also yields right variables';
+
+$branch      = "my/branch";
+$args_branch = "https://github.com/user/repo/tree/$branch https://openqa.opensuse.org/tests/1234 FOO=bar";
+$dirs
+  = "CASEDIR=https://github.com/user/repo.git#$branch PRODUCTDIR=repo/product NEEDLES_DIR=/my/case/dir/product/needles";
+$expected_branch_re
+  = qr{${clone_job}1234 _GROUP=0 TEST=my_test\@user/repo#$branch BUILD=user/repo#$branch ${dirs} FOO=bar};
+test_once $args_branch, $expected_branch_re, 'branch which contains slash';
+
```

Comment:
I think you can simply keep the old tests in place but replace "my_branch" with "my/branch".

Reply:
That makes sense, I'll do that.

Reply:
Now I think you don't even need the lines 68-74 anymore as this should test only what is done further above anyway, isn't it?

If you think we should keep it then please try to prevent the line breaks, e.g. extract "user/repo" into a variable $prj?

Reply:
Good point, I removed these lines. Now hopefully ready.